Input Autocomplete

KISS opened the pull request:
This is a copy of the `input-autocomplete` element in ACCESS Patterns: https://accesspatterns.cityofnewyork.us/inputs#autocomplete 

Things that were changed:
- using `input-auto..` vs `inputs-auto..`
- documentation has an `accessibility` section at the bottom
- `listItem()` bug re:  `const li = ...` conditional was fixed
- styles were changed to fit NYCO brand

Note: Holding off on other enhancements like setting a score threshold for dropdown list

The commits to review include `[Feat/Autocomplete]` in their name (last two)

KISS commented:
@kimpenguin yeah, this is one of the features that was left out for now (score threshold) and will require looking into how the score is calculated since there's a bug. by default it's set to only show 5 suggestions at a time so the number of suggestions is limited in that way.

devowhippit commented:
Looks good, I would comment out the accessibility note until we can revisit. @KISS is there a ticket for keeping track of the additional enhancements?

KISS commented:
@devowhippit I added a comment with this info to the `Accessible Autocomplete` ticket in JIRA (https://hhsconnect.atlassian.net/browse/PTTRN-28) and also added a comment in the related ticket for PD (https://hhsconnect.atlassian.net/browse/PD-102)
